prover/vk_setup_data_generator_server_fri/src/keystore.rs

```diff
@@ -412,6 +391,26 @@ impl Keystore {
         )
         .context("save_finalization_hints()")?;
 
+        // Recursion tip
+        self.save_recursive_layer_verification_key(source.get_recursion_tip_vk().map_err(
+            |err| anyhow::anyhow!("No vk exist for recursion tip layer circuit: {err}"),
+        )?)
+        .context("save_recursion_tip_vk")?;
+
+        let node_hint = source
+            .get_recursion_tip_finalization_hint()
+            .map_err(|err| {
+                anyhow::anyhow!("No finalization hint exist for recursion tip layer circuit: {err}")
+            })?
```

Comment:
Off-topic (?): Shame that `get_recursion_tip_finalization_hint()` doesn't return a type convertible to `anyhow::Error`; otherwise, it'd be possible to use `.context()` here.